backend/schema/core.ts

```diff
@@ -1,15 +1,21 @@
 export const coreProperties = {
-  weightInKg: {
-    type: "number",
-  },
-  lengthInCm: {
-    type: "number",
-  },
-  widthInCm: {
-    type: "number",
-  },
-  heightInCm: {
-    type: "number",
+  weight: {
+    type: "object",
+    properties: {
+      value: { type: "number" },
+      unit: { type: "string" },
+    },
+    required: ["value", "unit"],
+  },
+  dimensions: {
+    type: "object",
+    properties: {
+      length: { type: "number" },
+      width: { type: "number" },
+      height: { type: "number" },
+      unit: { type: "string" },
+    },
+    required: ["length", "width", "height", "unit"],
   },
```

Comment:
@rickycheah I have made changes to these fields.
- weight now returns back a value as well as a unit
- for length/width/height, its now embedded inside the dimensions object instead.

Can you please also make the corresponding changes in the front-end?